File: example/python/worker.py
#!/usr/bin/env python3
"""
Worker script to run the Temporal worker that executes workflows and activities.
"""

import asyncio
import logging
from temporalio import activity, workflow
from temporalio.client import Client
from temporalio.worker import Worker

# Import our workflow and activities
from workflow import (
    UserOnboardingWorkflow,
    fetch_user_data,
    validate_user_data, 
    send_welcome_email,
    create_user_profile
)

# Configure logging
logging.basicConfig(level=logging.INFO)
logger = logging.getLogger(__name__)

async def main():
    """Main function to start the Temporal worker"""
    
    # Connect to Temporal server (default localhost:7233)
    client = await Client.connect("localhost:7233")
    
    # Create worker
    worker = Worker(
        client,
        task_queue="user-onboarding-task-queue",
        workflows=[UserOnboardingWorkflow],
        activities=[
            fetch_user_data,
            validate_user_data,
            send_welcome_email,
            create_user_profile
        ]
    )
    
    logger.info("🚀 Starting Temporal worker for user onboarding...")
    logger.info("📋 Task Queue: user-onboarding-task-queue")
    logger.info("🔧 Registered Workflow: UserOnboardingWorkflow")
    logger.info("⚡ Registered Activities: fetch_user_data, validate_user_data, send_welcome_email, create_user_profile")
    logger.info("🔄 Worker is ready to process workflows and activities...")
    
    # Start the worker
    await worker.run()

if __name__ == "__main__":
    try:
        asyncio.run(main())
    except KeyboardInterrupt:
        logger.info("🛑 Worker stopped by user")
    except Exception as e:
        logger.error(f"❌ Worker failed: {e}")
        raise 

File: example/go/structured-workflow/temporal-cli/worker.go
package main

import (
	"log"

	"go.temporal.io/sdk/client"
	"go.temporal.io/sdk/worker"
)

func startWorker() error {
	// Create the Temporal client
	c, err := client.Dial(client.Options{
		HostPort: client.DefaultHostPort,
	})
	if err != nil {
		return err
	}
	defer c.Close()

	// Create a worker that listens on the specified task queue
	w := worker.New(c, TaskQueue, worker.Options{})

	// Register the workflow and activity functions
	w.RegisterWorkflow(ExampleWorkflow)
	w.RegisterActivity(ExampleActivity)

	log.Printf("Worker starting on task queue: %s", TaskQueue)
	log.Println("Available workflows:")
	log.Println("  - ExampleWorkflow")
	log.Println("Available activities:")
	log.Println("  - ExampleActivity")
	log.Println()
	log.Println("Worker is ready to process workflows and activities...")
	log.Println("Press Ctrl+C to stop the worker")

	// Start the worker and block until interrupted
	return w.Run(worker.InterruptCh())
}
